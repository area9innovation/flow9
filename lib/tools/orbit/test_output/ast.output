true
true
true
false
Result: 0
