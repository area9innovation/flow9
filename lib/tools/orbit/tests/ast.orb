fn is_number(expr : ast) = (astname(expr) == "Int" || astname(expr) == "Double");
fn is_var(expr : ast) = (astname(expr) == "UId" || astname(expr) == "Identifier");
fn is_glex(expr : ast) = (
	is_var(eval(expr)) || is_number(eval(expr)) ||
	expr is (
		a * b => is_glex(eval(a)) && is_glex(eval(b));
		a ^ b => (
			is_glex(eval(a)) && is_number(eval(b))
		);
		a => false
	)
);
fn quote(e : ast) = e;
println(is_glex((x^2)));
println(is_glex((a*x^2)));
println(is_glex((x^2*x^4*1*5)));
println(is_glex((a+b)));
