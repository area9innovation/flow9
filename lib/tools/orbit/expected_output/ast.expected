true
true
true
false
Result: false
